client/service/src/builder.rs

```diff
@@ -1074,7 +1074,8 @@ ServiceBuilder<
 					.unwrap_or(0),
 			);
 			if let Some(metrics) = metrics.as_ref() {
-				metrics.memory_usage_bytes.set(memory);
+				// `sysinfo::Process::memory` returns memory usage in KiB and not bytes.
+				metrics.memory_usage_bytes.set(memory * 1024);
```

Comment:
I would guess we can ignore the overflow error that would happen on machines with > 16 Exabytes of memory.